ownCloud/Client/Sharing/GroupSharingTableViewController.swift

```diff
@@ -0,0 +1,465 @@
+				} else {
+					shareRows.append( StaticTableViewRow(rowWithAction: nil, title: share.recipient!.displayName!, subtitle: share.permissionDescription(), accessoryType: .none) )
+				}
+			}
+			let sectionType = "share-section-\(String(type.rawValue))"
+			if let section = self.sectionForIdentifier(sectionType) {
+				self.removeSection(section)
+			}
+
+			let section : StaticTableViewSection = StaticTableViewSection(headerTitle: title, footerTitle: nil, identifier: sectionType, rows: shareRows)
+			self.addSection(section, animated: true)
+		}
+	}
+
+	func addShareSections() {
+		OnMainThread {
+			self.addSectionFor(type: .userShare, with: "Users".localized)
+			self.addSectionFor(type: .groupShare, with: "Groups".localized)
+			self.addSectionFor(type: .remote, with: "Remote Users".localized)
+		}
+	}
+
+	func removeShareSections() {
+		OnMainThread {
+			let types : [OCShareType] = [.userShare, .groupShare, .remote]
+			for type in types {
+				let identifier = "share-section-\(String(type.rawValue))"
+				if let section = self.sectionForIdentifier(identifier) {
+					self.removeSection(section)
+				}
+			}
+
+			let identifier = "action-section"
+			if let section = self.sectionForIdentifier(identifier) {
+				self.removeSection(section)
+			}
+		}
+	}
+
+	func resetTable(showShares : Bool) {
+		removeShareSections()
+		messageView?.message(show: false)
+		if let section = self.sectionForIdentifier("search-results") {
+			self.removeSection(section)
+		}
+		if shares.count > 0 && showShares {
+			self.addShareSections()
+		}
+	}
+
+	// MARK: - Sharing Helper
+
+	func share(at indexPath : IndexPath) -> OCShare? {
+		var type : OCShareType?
+		switch indexPath.section {
+		case 0:
+			type = .userShare
+		case 1:
+			type = .groupShare
+		case 2:
+			type = .remote
+		default:
+			break
+		}
+
+		if type != nil {
+			let shares = self.shares.filter { (OCShare) -> Bool in
+				if OCShare.type == type {
+					return true
+				}
+				return false
+			}
+			if shares.indices.contains(indexPath.row) {
+				return shares[indexPath.row]
+			}
+		}
+
+		return nil
+	}
+
+	func canEdit(share: OCShare) -> Bool {
+		if core?.connection.loggedInUser?.userName == share.owner?.userName || core?.connection.loggedInUser?.userName == share.itemOwner?.userName {
+			return true
+		}
+
+		return false
+	}
+
+	// MARK: - UISearchResultsUpdating Delegate
+
+	func updateSearchResults(for searchController: UISearchController) {
+		guard let text = searchController.searchBar.text else { return }
+		if text.count > core?.connection.capabilities?.sharingSearchMinLength?.intValue ?? 1 {
+			recipientSearchController?.searchTerm = text
+		} else if searchController.isActive {
+			resetTable(showShares: false)
+		}
+	}
+
+	func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+		self.resetTable(showShares: true)
+		self.searchController?.searchBar.isLoading = false
+	}
+
```

Comment:
Adding this code snippet will pre-populate with suggestions from the server when the search bar is focused the first time:
```swift

	func searchBarShouldBeginEditing(_ searchBar: UISearchBar) -> Bool {
		if recipientSearchController?.recipients == nil, recipientSearchController?.searchTerm == nil {
			recipientSearchController?.search()
		}
		return true
	}
```
<img width="515" alt="Bildschirmfoto 2019-05-17 um 16 09 45" src="https://user-images.githubusercontent.com/639669/57933676-316f2e00-78be-11e9-8e21-9aa33c9dd4eb.png">

Reply:
This may be not wanted.